backend/src/main/java/gov/cdc/usds/simplereport/service/UploadService.java

```
@@ -0,0 +1,96 @@
+    }
+
+    public void processPersonCSV(InputStream csvStream) throws IOException {
+        final MappingIterator<Map<String, String>> valueIterator = new CsvMapper()
+                .enable(CsvParser.Feature.FAIL_ON_MISSING_COLUMNS)
+                .readerFor(Map.class)
+                .with(PERSON_SCHEMA)
+                .readValues(csvStream);
+
+        // Since the CSV parser won't fail when give a single string, we simple check to see if it has any parsed values
+        // If not, we throw an error assuming the user didn't actually want to submit something empty.
+        if (!valueIterator.hasNext()) {
+            throw new IllegalArgumentException("Empty or invalid CSV submitted");
+        }
+
+        while (valueIterator.hasNext()) {
+            final Map<String, String> row = valueIterator.next();
+
+            final LocalDate patientDOB = LocalDate.parse(row.get("DOB"));
+            _ps.addPatient(row.get("ID"),
+                    row.get("FirstName"),
+                    row.get("MiddleName"),
+                    row.get("LastName"),
+                    row.get("Suffix"),
+                    patientDOB,
+                    row.get("Street"),
+                    row.get("Street2"),
+                    row.get("City"),
+                    row.get("State"),
+                    row.get("ZipCode"),
+                    row.get("County"),
+                    row.get("PhoneNumber"),
+                    row.get("typeOfHealthcareProfessional"),
+                    row.get("Email"),
+                    row.get("Race"),
+                    row.get("Gender"),
+                    row.get("Ethnicity"),
+                    Boolean.parseBoolean(row.get("residentCongregateSetting")),
+                    Boolean.parseBoolean(row.get("employedInHealthcare")));
+        }
+    }
+
+    private static CsvSchema personSchema() {
+        return CsvSchema.builder()
+                .addColumn("ID", CsvSchema.ColumnType.STRING)
```

It kind of kills me how many in-line string constants this is. In a better world we'd probably want an `enum` to map the logical field names to the physical field names. Seems slightly dubious for today, but if we start editing header names I would recommend it strongly (though hopefully we'd get errors rather than silent failures if we had a copy/paste error?)